clients/csharp-uwp/UWPVoiceAssistantSample/AudioOutput/DialogAudioOutputAdapter.cs

```diff
@@ -207,12 +206,11 @@ private async Task RegenerateAudioGraphAsync()
                 this.graph = creationResult.Graph;
 
                 this.frameInputNode = this.graph.CreateFrameInputNode(AudioEncodingProperties.CreatePcm(16000, 1, 16));
-                this.frameInputNode.QuantumStarted += this.FrameInputNode_QuantumStarted;
+                this.frameInputNode.QuantumStarted += this.OnFrameInputQuantumStarted;
                 this.frameInputNode.AudioFrameCompleted += async (s, e) =>
                 {
-                    this.activeUtterances -= 1;
-
-                    if (this.audioOutputStreams.Count == 0 && this.activeUtterances == 0)
+                    // If we just finished processing an audio frame and there's no more data to process, output is done.
```

Comment:
This event is constantly fired while the graph is running. The if statement is triggered when we dequeue in line 297. This whole event handler essentially functions as a background task with this code:

while (this.audioOutputStreams.Count != 0) { // wait }
await this.StopPlaybackAsync();

How about we move this if statement to after line 297? The logic would be: wait for the current stream to finish, then dequeue this stream. If nothing else has been added to the queue while this stream was playing, then stop playback.

Reply:
Per the talk we had, I expanded the PR (substantially) to just get rid of the now-redundant prompt queue to hopefully work around this. I took the recommendation of signaling the "end" inside of the QuantumStarted handler (at the end of it, after the AddFrame call). I think this _might_ still end up signalling the OutputEnded event a portion of a single quantum early, but with quanta now 10ms rather than up to 160,000ms I think that should be manageable.